commons-statistics-descriptive/src/test/java/org/apache/commons/statistics/descriptive/SumTest.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.statistics.descriptive;
+
+import java.util.Arrays;
+import java.util.stream.Stream;
+import org.apache.commons.rng.UniformRandomProvider;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/**
+ * Test for {@link Sum}.
+ */
+final class SumTest {
+
+    @Test
+    void testEmpty() {
+        Sum sum = Sum.create();
+        Assertions.assertEquals(0.0, sum.getAsDouble());
+    }
+
+    @Test
+    void testNan() {
+        Sum sum = Sum.create();
+        double[] testArray = {Double.NaN, +0.0d, -0.0d, Double.POSITIVE_INFINITY, Double.NEGATIVE_INFINITY};
+        for (double value : testArray) {
+            sum.accept(value);
+        }
+        Assertions.assertEquals(Double.NaN, sum.getAsDouble());
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "testSum")
+    void testSum(double[] values, double expected) {
+        Sum sum = Sum.create();
+        for (double value : values) {
+            sum.accept(value);
+        }
+        Assertions.assertEquals(expected, sum.getAsDouble(), "sum");
+        Assertions.assertEquals(expected, Sum.of(values).getAsDouble(), "of (values)");
+    }
+
+    static Stream<Arguments> testSum() {
```

Comment:
This could use some extended precision examples. This [one](https://en.wikipedia.org/wiki/Kahan_summation_algorithm#Further_enhancements) is useful:
```
1, 10e100, 1, -10e100 == 2
```

Reply:
I had included a similar test case as the one above in my [PR](https://github.com/ani5rudh/commons-statistics/blob/846c6185330439a4bb7b1df25611dd3ddcd696ed/commons-statistics-descriptive/src/test/java/org/apache/commons/statistics/descriptive/SumTest.java#L105).